core/src/main/java/com/datastax/oss/driver/api/core/config/DriverConfigLoader.java

```diff
@@ -92,7 +116,22 @@ static DriverConfigLoader fromClasspath(@NonNull String resourceBaseName) {
    */
   @NonNull
   static DriverConfigLoader fromPath(@NonNull Path file) {
-    return fromFile(file.toFile());
+    return fromPath(file, DRIVER_CLASS_LOADER);
+  }
+
+  /**
+   * Just like {@link #fromPath(java.nio.file.Path)} except that the loader will use the provided
+   * {@code ClassLoader}, as opposed to the Driver's {@code ClassLoader}, to find the {@code Path}.
+   *
+   * <p>This is intended for use when an application's ClassLoader is different from the Driver's
+   * ClassLoader, and the current Context ClassLoader is not the same CLassLoader that loaded the
+   * application (eg. OSGi bundles). For web deployments not using OSGi, use of this method is
+   * normally not necessary, as the Context ClassLoader in that environment can usually access
+   * application bundled resources.
+   */
+  @NonNull
+  static DriverConfigLoader fromPath(@NonNull Path file, ClassLoader appClassLoader) {
```

Comment:
For `Path`, `File` and `URL` you don't need a class loader. The contents are retrieved by other mechanisms that do not depend on the class loader being used, so I think the overloaded methods are not useful.

Reply:
AIUI, the problem is that the fallback default configuration isn't found - so this needs to be fixed anywhere defaultReference() is called to load the default config, regardless of how the 'overriding' configuration is loaded.